presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -30,41 +37,70 @@
     /**
      * Returns if the catalog with the given name is available in metadata.
      */
-    boolean catalogExists(String catalogName);
+    default boolean catalogExists(String catalogName)
```

Comment:
why do these now have a default implementation?

Reply:
The idea is to reduce the boilerplate code in the AbstractMockMetadata. I do not have strong preference either way.

Reply:
seems kind of risky that someone could accidentally not include one of these definitions in a real implementation.  I would at least throw an exception for all these cases instead of returning some value.

Reply:
Okay, let me remove default implementation.